.changeset/odd-waves-laugh.md

```diff
@@ -0,0 +1,11 @@
+---
+'xstate': minor
+---
+
+Send an event to multiple targets by providing a list of references for a _send_ action creator's `.to` property: E.g.:
+
+```js
+send('SOME_EVENT', to: ['actor1', 'invokedMachineId2'])
+...
+send('SOME_EVENT', to: (ctx, evt) => [ctx.actor1, ctx.actor2, ...evt.addtionalTargets])
```

Comment:
I have a concern that this is really dynamic and we might not be able to convert this pattern to SCXML. We can't know upfront (statically) how many send actions this would return.

I'm wondering if this problem shouldn't really be solved in userland (or with a helper provided by us):
```js
const sendToMany = (event, { to: targets, ...options }) =>
  targets.map((to) => send(event, { to, ...options }));
```
This would solve the "static" list of targets, whereas the dynamic list of targets should most likely be solved by `forEach` action which we plan to introduce (SCXML has support for this). You'd need to keep all your targets in a single array-collection for `forEach` to handle this but this seems like a good constraint - forcing you to organize your context values.

I recognize the need to cover this use case, I'm just proposing alternative solutions that wouldn't suffer from the mentioned drawback